src/node/encryptor.h

```diff
@@ -85,10 +88,33 @@ namespace ccf
 
     std::list<EncryptionKey> encryption_keys;
 
-    void set_iv(crypto::GcmHeader<crypto::GCM_SIZE_IV>& gcm_hdr)
+    void check_iv_validity(kv::Version version, View view)
     {
-      gcm_hdr.set_iv_id(id);
-      gcm_hdr.set_iv_seq(seqNo.fetch_add(1));
+      // expects iv to be locked
+      if (version <= last_used_version)
+      {
+        if (view <= last_used_view)
+        {
+          throw std::logic_error(fmt::format(
```

Comment:
Should we safely recovery from this?

Reply:
How would you propose we recover?

Reply:
Will remove this check. Will keep the original implementation for Raft and a view + version implementation for PBFT. It should be impossible for PBFT to rollback without a view change, so versions should not be re-used with the same view number. Hence this check is redundant.